sauron/sauron.py

```diff
@@ -6,7 +6,8 @@
 from pyln.client import Plugin
 
 
-plugin = Plugin()
+plugin = Plugin(dynamic=False)
+SESSION = requests.session()
```

Comment:
Notice that adding a session here means that any cookie set by the server is sent back on subsequent requests, which might allow a server to collate the requests easily.

Reply:
Yeah, but if I go this way I should also renew circuits after each call to `gettxout`, send tx, get tx as the exit node ip might allow to collate the requests too ?

Reply:
This is not viable so I thought about some heuristics (keeping track of txids, timeouts), in the same vein maybe drop `requests` for a more low level solution (pycurl?) to avoid automagically leaking data.
At the end I considered this as a "avoid opening a public port and do port forwarding for my home Esplora instance" more than a solution to hit public APIs and added the above comment to the README..

I can still renew session though.
Do you think we should consider the run-against-public-APIs case ?

Reply:
I think it's ok, but we can generate a new session on each distinct action. But afaik Tor circuits are persistent for a time / destination, so not much to be gained here.